packages/react-native/ReactCommon/react/performance/timeline/PerformanceObserverRegistry.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "PerformanceObserverRegistry.h"
+#include "PerformanceObserver.h"
+
+namespace facebook::react {
+
+void PerformanceObserverRegistry::addObserver(
+    std::shared_ptr<PerformanceObserver> observer) {
+  std::lock_guard guard(observersMutex_);
+  observers_.insert(observer);
+}
+
+void PerformanceObserverRegistry::removeObserver(
+    std::shared_ptr<PerformanceObserver> observer) {
+  std::lock_guard guard(observersMutex_);
+  observers_.erase(observer);
+}
+
+void PerformanceObserverRegistry::queuePerformanceEntry(
+    const PerformanceEntry& entry) {
+  std::lock_guard lock(observersMutex_);
+
+  // filter dead observers
```

Comment:
We don't need this logic, and we can get rid of the weak pointer / ownerless logic. Observers should only be GC'ed when the value in JS is disposed after disconnecting.